src/spy.js

```diff
@@ -0,0 +1,20 @@
+export function spy(cb) {
+	const calls = [];
+	const wrapped = function(...args) {
+		const call = { args, return: undefined };
+		calls.push(call);
+		const res = cb.apply(this, args)
+
+		if (res && typeof res.then === "function") {
+			return res.then(r => {
+				call.return = r;
+				return r;
+			})
+		}
+
+		call.return = res;
+		return res;
+	};
+	wrapped.calls = calls;
+	return wrapped;
```

Comment:
Some functions rely on that `length` is set correctly (like if you are working with `curry`). 

```ts
const myFunction = (a, b) => a + b;
myFunction.length // 2
fn = spy(myFunction);
fn.length // 0
```

```suggestion
  Object.defineProperty(wrapped, 'length', { value: cb.length });
	return wrapped;
```